recipes/yaml-cpp/all/conanfile.py

```diff
@@ -23,18 +26,27 @@ def _source_subfolder(self):
         return "source_subfolder"
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = self.name + "-" + self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
 
     def config_options(self):
         if self.settings.os == 'Windows':
             del self.options.fPIC
 
     def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
         if self.settings.compiler.cppstd:
             tools.check_min_cppstd(self, "11")
 
+    def validate(self):
+        if self.settings.compiler == "Visual Studio" and self.options.shared and self.settings.compiler.runtime == "MTd":
```

Comment:
Usually we check for a static runtime with `"MT" in self settings.compiler.runtime`

Reply:
Thanks for the clarification.

> Usually we check for a static runtime with `"MT" in self settings.compiler.runtime`

Yes, I have seen this approach. I tried MT and it worked fine locally and on CI (as opposed to MTd). Probably better to ban both MT and MTd.

Reply:
If it fails with `MTd` but succeeds with `MT`, then it feels like it's a bug or we're entering undefined behavior land.
I'm thinking it's the latter so would just disable shared and MT.

The C++ allocator will not be shared between exe and sell, which will cause problems.

Reply:
If it fails with `MTd` but succeeds with `MT`, then it feels like it's a bug or we're entering undefined behavior land.
I'm thinking it's the latter so would just disable shared and MT.

The C++ allocator will not be shared between exe and dll, which will cause problems.

Reply:
Yes, better safe than sorry. Disabled it.